core/shared/src/main/scala/fs2/text.scala

```diff
@@ -360,12 +383,25 @@ object text {
           chunk.foreach { string =>
             fillBuffers(stringBuilder, linesBuffer, string)
           }
-          Pull.output(Chunk.buffer(linesBuffer)) >> go(stream, stringBuilder, first = false)
+
+          maxLineLength match {
```

Comment:
There is `LinesBenchmark`, can you, please, run it against `main` version and this one?